File: COMP303_FinalTest/bank-frontend/src/services/bankService.js
import axios from './axiosConfig';

const API_URL = '/api/banks';

// Add a new bank
const addBank = (bank) => {
    return axios.post(API_URL, bank);
};

// Get a bank by ID
const getBankById = (id) => {
    return axios.get(`${API_URL}/id/${id}`);
};

// Get a bank by name
const getBankByName = (name) => {
    return axios.get(`${API_URL}/name/${encodeURIComponent(name)}`);
};

// Get all banks
const getAllBanks = () => {
    return axios.get(API_URL);
};

// Delete a bank by ID
const deleteBankById = (id) => {
    return axios.delete(`${API_URL}/id/${id}`);
};

// Delete a bank by name
const deleteBankByName = (name) => {
    return axios.delete(`${API_URL}/name/${encodeURIComponent(name)}`);
};

// Update a bank by ID
const updateBankById = (id, bank) => {
    return axios.put(`${API_URL}/id/${id}`, bank);
};

// Update a bank by name
const updateBankByName = (name, bank) => {
    return axios.put(`${API_URL}/name/${encodeURIComponent(name)}`, bank);
};

export default {
    addBank,
    getBankById,
    getBankByName,
    getAllBanks,
    deleteBankById,
    deleteBankByName,
    updateBankById,
    updateBankByName,
};

File: COMP303_FinalTest/src/main/java/com/Harshbir/FinalTest_COMP303/service/BankService.java
package com.Harshbir.FinalTest_COMP303.service;

import com.Harshbir.FinalTest_COMP303.entity.Bank;
import com.Harshbir.FinalTest_COMP303.repository.BankRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class BankService {

    @Autowired
    private BankRepository bankRepository;

    // Add a new bank
    public Bank addBank(Bank bank) {
        return bankRepository.save(bank);
    }

    // Get a bank by ID
    public Optional<Bank> getBankById(Long id) {
        return bankRepository.findById(id);
    }

    // Get a bank by name
    public Optional<Bank> getBankByName(String name) {
        return bankRepository.findByBankName(name);
    }

    // Get all banks
    public List<Bank> getAllBanks() {
        return bankRepository.findAll();
    }

    // Delete a bank by ID
    public void deleteBankById(Long id) {
        bankRepository.deleteById(id);
    }

    // Delete a bank by name
    public void deleteBankByName(String name) {
        bankRepository.deleteByBankName(name);
    }

    // Update a bank by ID
    public Bank updateBankById(Long id, Bank updatedBank) {
        return bankRepository.findById(id).map(bank -> {
            bank.setBankName(updatedBank.getBankName());
            bank.setBankYear(updatedBank.getBankYear());
            bank.setBankEmp(updatedBank.getBankEmp());
            bank.setBankAddress(updatedBank.getBankAddress());
            bank.setBankBranches(updatedBank.getBankBranches());
            bank.setBankATMs(updatedBank.getBankATMs());
            return bankRepository.save(bank);
        }).orElseThrow(() -> new RuntimeException("Bank not found with ID " + id));
    }

    // Update a bank by name
    public Bank updateBankByName(String name, Bank updatedBank) {
        return bankRepository.findByBankName(name).map(bank -> {
            bank.setBankName(updatedBank.getBankName());
            bank.setBankYear(updatedBank.getBankYear());
            bank.setBankEmp(updatedBank.getBankEmp());
            bank.setBankAddress(updatedBank.getBankAddress());
            bank.setBankBranches(updatedBank.getBankBranches());
            bank.setBankATMs(updatedBank.getBankATMs());
            return bankRepository.save(bank);
        }).orElseThrow(() -> new RuntimeException("Bank not found with name " + name));
    }
}